sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs

```diff
@@ -0,0 +1,80 @@
+
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.HardwareSecurityModules.Models;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.HardwareSecurityModules
+{
+    /// <summary>
+    /// A class representing the DedicatedHsm data model.
+    /// Resource information with extended details.
+    /// </summary>
+    public partial class DedicatedHsmData : TrackedResourceData
+    {
+        /// <summary> Initializes a new instance of DedicatedHsmData. </summary>
+        /// <param name="location"> The location. </param>
+        public DedicatedHsmData(AzureLocation location) : base(location)
+        {
+            Zones = new ChangeTrackingList<string>();
+        }
+
+        /// <summary> Initializes a new instance of DedicatedHsmData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="tags"> The tags. </param>
+        /// <param name="location"> The location. </param>
+        /// <param name="networkProfile"> Specifies the network interfaces of the dedicated hsm. </param>
+        /// <param name="managementNetworkProfile"> Specifies the management network interfaces of the dedicated hsm. </param>
+        /// <param name="stampId"> This field will be used when RP does not support Availability zones. </param>
+        /// <param name="statusMessage"> Resource Status Message. </param>
+        /// <param name="provisioningState"> Provisioning state. </param>
+        /// <param name="sku"> SKU details. </param>
+        /// <param name="zones"> The Dedicated Hsm zones. </param>
+        internal DedicatedHsmData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, IDictionary<string, string> tags, AzureLocation location, NetworkProfile networkProfile, NetworkProfile managementNetworkProfile, string stampId, string statusMessage, JsonWebKeyType? provisioningState, HardwareSecurityModulesSku sku, IList<string> zones) : base(id, name, resourceType, systemData, tags, location)
+        {
+            NetworkProfile = networkProfile;
+            ManagementNetworkProfile = managementNetworkProfile;
+            StampId = stampId;
+            StatusMessage = statusMessage;
+            ProvisioningState = provisioningState;
+            Sku = sku;
+            Zones = zones;
+        }
+
+        /// <summary> Specifies the network interfaces of the dedicated hsm. </summary>
+        public NetworkProfile NetworkProfile { get; set; }
+        /// <summary> Specifies the management network interfaces of the dedicated hsm. </summary>
+        public NetworkProfile ManagementNetworkProfile { get; set; }
+        /// <summary> This field will be used when RP does not support Availability zones. </summary>
+        public string StampId { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

Reply:
StampId is just a string like "1" that our RP will use when the region does not support availability zones

Reply:
Got it. Thanks.